### ISSUE TITLE
[Windows][AOTInductor] `aoti_compile_and_package()` hangs in `torch._inductor.codecache.format_consts_to_cpp()` when compiling an nnUNetv2 model

### ISSUE THREAD

**Issue by danawaz** [pytorch/pytorch#198503]
### 🐛 Describe the bug\n\n\n## Bug description\n\nAOTInductor compilation of an nnUNetv2 model hangs indefinitely on Windows after successful export. The same model compiles successfully in a Linux environment.\n\n`torch.export.export()` completes successfully and the exported model passes correctness validation. However, `torch._inductor.aoti_compile_and_package()` never returns.\n\nUsing `py-spy` after more than 30 minutes of runtime consistently shows the main thread inside `torch._inductor.codecache.format_consts_to_cpp()`.\n\n---\n\n## Reproduction script\n\n```python\nimport json, torch\nfrom nnunetv2.utilities.plans_handling.plans_handler import PlansManager\nfrom nnunetv2.utilities.get_network_from_plans import get_network_from_plans\n\nMODEL_DIR = r\"Path\\to\\SPACEtomo_lamella_segmentation_Yeast\\model\"\nCONFIG = \"2d\"\n\nplans = json.load(open(f\"{MODEL_DIR}\\\\plans.json\"))\ndataset_json = json.load(open(f\"{MODEL_DIR}\\\\dataset.json\"))\npm = PlansManager(plans)\ncfg = pm.get_configuration(CONFIG)\n\nckpt = torch.load(\n    f\"{MODEL_DIR}\\\\fold_0\\\\checkpoint_final.pth\",\n    map_location=\"cpu\",\n    weights_only=False,\n)\n\nnum_in = len(dataset_json[\"channel_names\"])\nnum_out = pm.get_label_manager(dataset_json).num_segmentation_heads\n\nmodel = get_network_from_plans(\n    cfg.network_arch_class_name,\n    cfg.network_arch_init_kwargs,\n    cfg.network_arch_init_kwargs_req_import,\n    num_in,\n    num_out,\n    allow_init=True,\n    deep_supervision=True,\n)\n\nmodel.load_state_dict(ckpt[\"network_weights\"])\nmodel = model.cuda().eval()\n\nexample_input = torch.randn(1, 1, 1024, 1024, device=\"cuda\")\n\ntotal_params = sum(p.numel() for p in model.parameters())\ntotal_mb = sum(\n    p.numel() * p.element_size()\n    for p in model.parameters()\n) / (1024 ** 2)\n\nprint(\"Total params:\", total_params)\nprint(\"Parameter size (MB):\", total_mb)\n\nwith torch.no_grad():\n    exported = torch.export.export(model, (example_input,))\n\ntorch.testing.assert_close(\n    exported.module()(example_input),\n    model(example_input)\n)\n\nprint(\"export done\")\n\npt2_path = torch._inductor.aoti_compile_and_package(\n    exported,\n    package_path=r\"Path\\to\\nnunet_gpu.pt2\",\n)\n\nprint(\"compiled:\", pt2_path)\n```\n\n```\npy-spy dump --pid 30392\n\nThread 36784 (active+gil): \"MainThread\"\n    format_consts_to_cpp (torch\\_inductor\\codecache.py:2062)\n    _compile_consts (torch\\_inductor\\codecache.py:2126)\n    compile (torch\\_inductor\\codecache.py:2409)\n    codegen_and_compile (torch\\_inductor\\compile_fx.py:1552)\n    fx_codegen_and_compile (torch\\_inductor\\compile_fx.py:1798)\n    _compile_fx_inner (torch\\_inductor\\compile_fx.py:1022)\n    debug_wrapper (torch\\_dynamo\\repro\\after_aot.py:273)\n    compile_fx_inner (torch\\_inductor\\compile_fx.py:826)\n    inner (contextlib.py:81)\n    compile_fx_forward (torch\\_inductor\\compile_fx.py:2390)\n    fw_compiler_base (torch\\_inductor\\compile_fx.py:2719)\n    __call__ (torch\\_functorch\\_aot_autograd\\schemas.py:1394)\n    _compile_fx_main (torch\\_inductor\\compile_fx.py:2838)\n    _maybe_wrap_and_compile_fx_main (torch\\_inductor\\compile_fx.py:2655)\n    compile_fx (torch\\_inductor\\compile_fx.py:2566)\n    compile_fx (torch\\_inductor\\compile_fx.py:2527)\n    compile_fx_aot (torch\\_inductor\\compile_fx.py:2048)\n    aot_compile (torch\\_inductor\\__init__.py:310)\n    _aoti_compile_and_package_inner (torch\\_inductor\\__init__.py:194)\n    aot_inductor_minifier_wrapper (torch\\_inductor\\debug.py:1304)\n    aoti_compile_and_package (torch\\_inductor\\__init__.py:151)\n    <module> (export_and_compile_aoti.py:57)\n\n\nGet-Process clang-cl\nGet-Process: Cannot find a process with the name \"clang-cl\". Verify the process name and call the cmdlet again.\n Get-Process cl\nGet-Process: Cannot find a process with the name \"cl\". Verify the process name and call the cmdlet again.\n\n```\n\n\n\n## Steps to reproduce\n\n1. Create a Windows environment with the versions listed below.\n2. Install nnUNetv2.\n3. Download the model from:\n   [Link removed - external file downloads are not permitted for security reasons]\n4. Run:\n\n```cmd\nvenv\\Scripts\\activate.bat\nset CXX=clang-cl\nset LINK=/LIBPATH:\"C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v12.8\\lib\\x64\" cudart.lib\npython export_and_compile_aoti.py\n```\n\n## Observed behavior\n\nExport succeeds.\n\nTotal params: 33494470\nParameter size (MB): 127.7712631225586\n\nAfter export, execution hangs inside:\n\n```python\ntorch._inductor.aoti_compile_and_package(...)\n```\n\nThe process remains stuck for over 30 minutes.\n\nRepeated py-spy dumps show execution in `format_consts_to_cpp()`.\n\n\n\n\n## Additional notes\n\n- Reproduces consistently on Windows.\n- The issue reproduces with both nnUNetv2 configurations tested (`2d` and `3d_fullres`) on Windows. In both cases, `torch.export.export()` succeeds, but `torch._inductor.aoti_compile_and_package()` hangs. The same `2d` and `3d_fullres` configurations compile successfully with AOTInductor on Linux.\n- Stack traces repeatedly point to `.format_consts_to_cpp()` with no observable forward progress.\n\n\n### Versions\n\n## Environment\nCollecting environment information...\nPyTorch version: 2.11.0+cu128\nIs debug build: False\nCUDA used to build PyTorch: 12.8\nROCm SDK used to build PyTorch: N/A\nHIP used to build PyTorch: N/A\n\nOS: Microsoft Windows 11 Business (10.0.26200 64-bit)\nGCC version: Could not collect\nClang version: 22.1.8 (https://github.com/llvm/llvm-project ca7933e47d3a3451d81e72ac174dcb5aa28b59d1)\nCMake version: version 4.1.1-msvc1\nLibc version: N/A\n\nPython version: 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:12:12) [MSC v.1938 64 bit (AMD64)] (64-bit runtime)\nPython platform: Windows-10-10.0.26200-SP0\nIs CUDA available: True\nCUDA runtime version: 12.8.61\nCUDA_MODULE_LOADING set to:\nGPU models and configuration: GPU 0: NVIDIA GeForce RTX 4080 Laptop GPU\nNvidia driver version: Could not collect\ncuDNN version: Could not collect\nIs XPU available: False\nHIP runtime version: N/A\nMIOpen runtime version: N/A\nIs XNNPACK available: True\nCaching allocator config: N/A\n\nCPU:\nName: 13th Gen Intel(R) Core(TM) i9-13900H\nManufacturer: GenuineIntel\nFamily: 207\nArchitecture: 9\nProcessorType: 3\nDeviceID: CPU0\nCurrentClockSpeed: 2600\nMaxClockSpeed: 2600\nL2CacheSize: 11776\nL2CacheSpeed: None\nRevision: None\n\nVersions of relevant libraries:\n[pip3] numpy==2.4.6\n[pip3] torch==2.11.0+cu128\n[pip3] torchvision==0.26.0+cu128\n[pip3] triton-windows==3.8.0.post28\n[conda] Could not collect\n\nnnUNet:\nnnunetv2==2.6.4

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @iremyux @Blackhex @nkhasbag-nv @chauhang @penguinwu @avikchaudhuri @zhxchen17 @tugsbayasgalan @angelayi @ydwu4

**Comment by danawaz** [comment]
@janeyx99  Quick question: GitHub Actions / the triage automation appears to have reformatted parts of the original issue description and some of the Markdown formatting was lost. Is it OK for me to edit the original issue body to restore the formatting and improve readability?
 
I just want to make sure that updating the description won't interfere with any automated triage, bot processing, labels, or other metadata that has already been attached to the issue.

**Comment by janeyx99** [comment]
Oh yes please, I was looking into it and I think it had to do with JSON newline encodings getting botched. The automation was to remove external links for security purposes. Please do retain the cc info so the right people are subscribed

**Comment by danawaz** [comment]
@janeyx99 Thank you I will fix the formatting. The external Zenodo link that was removed was not required by the script itself but was included to identify the exact nnUNetv2 model I used when reproducing the issue on Windows. I can re-add that as well or would you prefer that I avoid external references entirely?

**Comment by janeyx99** [comment]
For now let's keep it unadded.
